src/XIVLauncher.Core/ConfigStorage.cs

```diff
@@ -0,0 +1,81 @@
+namespace XIVLauncher.Core;
+
+public class ConfigStorage
+{
+    public DirectoryInfo XLConfigRoot { get; }
+    public DirectoryInfo GameConfigRoot { get; }
+
+    private readonly DirectoryInfo legacyXLConfigRoot;
+    private readonly DirectoryInfo legacyGameConfigRoot;
+    private readonly DirectoryInfo symlinkGameConfigs;
+    
+    public ConfigStorage(string appName, string configFolder)
+    {
+        // https://developers.redhat.com/blog/2018/11/07/dotnet-special-folder-api-linux
+        // Support for migrating old configs, to bring storage inline with Windows
+        this.legacyXLConfigRoot = new DirectoryInfo(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), $".{appName}"));
+        this.legacyGameConfigRoot = new DirectoryInfo(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), $".{appName}/ffxivConfig"));
```

Comment:
I don't think we actually need this, seeing as xlcore isn't released yet and this stuff has been on master for less than a week... Better to make this less complex.